rotkehlchen/chain/evm/decoding/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,117 @@
+            deposit_event: bytes,
+            withdrawal_event: Optional[bytes],  # withdrawal is currently unsupported on gnosis
+            bridge_address: ChecksumEvmAddress,
+            bridged_asset: 'Asset',
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.bridged_asset = bridged_asset
+        self.deposit_event = deposit_event
+        self.withdrawal_event = withdrawal_event
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes a bridging event for `bridged_asset` specified.
+        Either a deposit or a withdrawal.
+        """
+        if context.tx_log.topics[0] not in {self.deposit_event, self.withdrawal_event}:
+            return DEFAULT_DECODING_OUTPUT
+
+        if context.tx_log.topics[0] == self.deposit_event:
+            from_address = context.transaction.from_address
+            to_address = self.bridge_address
+        else:
+            from_address = self.bridge_address
+            to_address = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.bridged_asset.resolve_to_crypto_asset(),
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_events=(self.deposit_event,),
+            source_chain=self.source_chain,
+            target_chain=self.target_chain,
+            from_address=from_address,
+            to_address=to_address,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == expected_location_label and
+                event.address == self.bridge_address and
+                event.asset == self.bridged_asset and
+                (amount is None or event.balance.amount == amount)
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=context.transaction.from_address,
+                    to_address=hex_or_bytes_to_address(context.tx_log.data[0:32]),
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=event.balance.amount,
+                    asset=self.bridged_asset.resolve_to_asset_with_symbol(),
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=XDAI_BRIDGE_CPT_DETAILS,
+                )
```

Comment:
here you should break if it's found. Also add an else clause that logs an error if the decoder could not find the matching transfer. There is many other decoders we do this in.
```suggestion
                )
                break
        else:
            log.error('')
```